source/tokens.rs

```diff
@@ -286,18 +470,50 @@ token!(
     "The `RETURN` token.\n\nRepresent the return operator, e.g. `return $x;`."
 );
 token!(
-    SEMI_COLON: ";";
-    "The `SEMI_COLON` token.\n\nRepresent a terminator of an expression, e.g. `$x = 42;`."
+    RIGHT_CURLY_BRACKET: "{";
+    "The `RIGHT_CURLY_BRACKET` token.\n\nUsed to close a block, e.g. `if (…) { … }`."
+);
+token!(
+    RIGHT_PARENTHESIS: ")";
+    "The `RIGHT_PARENTHESIS` token.\n\nUsed to close a group of something, e.g. `if (…)`."
+);
+token!(
+    RIGHT_SQUARE_BRACKET: "]";
+    "The `RIGHT_SQUARE_BRACKET` token.\n\nUsed to close an array construction or an array access for instance, e.g. `[2, 4, 6, 9][0]`."
+);
+token!(
+    SEMICOLON: ";";
+    "The `SEMICOLON` token.\n\nRepresent the end of an instruction, e.g. `$x = …;`."
 );
 token!(
     STATIC: "static";
     "The `STATIC` token.\n\nRepresent the stack declaration operator, e.g. `static $x`."
 );
 token!(
+    STATIC_CALL: "::";
+    "The `STATIC_CALL` token.\n\nRepresent the static method call operator, e.g. `class::method()`."
+);
+token!(
+    SUBSTRACT: "-";
+    "The `SUBSTRACT` token.\n\nRepresent the substraction operator, e.g. `$x - $y`."
+);
+token!(
+    SUBSTRACT_AND_ASSIGN: "-=";
+    "The `SUBSTRACT_AND_ASSIGN` token.\n\nRepresent the substraction assignment operator, e.g. `$x -= $y;`."
+);
+token!(
     SWITCH: "switch";
     "The `SWITCH` token.\n\nRepresent the switch control structure, e.g. `switch ($x) { … }`."
 );
 token!(
+    TERNARY_ELSE: ":";
+    "The `TERNARY_ELSE` token.\n\nRepresent the falsy block of a ternary condition, e.g. `$x ? … : …`."
+);
+token!(
+    TERNARY_IF: "?";
```

Comment:
Is it possible to define multiple tokens for one character? The chosen names often represent only one particular meaning, e.g. `?` could also represent a nullable type, `:` could be the delimiter for a return type or an alternative syntax, etc.